### PR TITLE
[datasets] new, listview (react)

### PR DESCRIPTION
**Pull request by nytai**
### CATEGORY

Choose one

- [ ] Bug Fix
- [ ] Enhancement (new features, refinement)
- [x] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY

Adds a new react listview to tables (soon to be renamed to datasets).

similar to https://github.com/apache/incubator-superset/pull/8845 and https://github.com/apache/incubator-superset/pull/8999

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

Screen Shot 2020-03-10 at 2 08 25 PM

Screen Shot 2020-03-10 at 2 08 49 PM

### TEST PLAN

- New listview (behind config `ENABLE_REACT_CRUD_VIEWS`) provides the same functionality as previous MVC crud views
- New listview is behind flag `ENABLE_REACT_CRUD_VIEWS`

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [x] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/connectors/sqla/models.py [hunk 1]
```diff
@@ -440,6 +440,18 @@ def make_sqla_column_compatible(
     def __repr__(self):
         return self.name
 
+    @property
+    def changed_by_name(self):
```
Comment (villebro) [inline superset/connectors/sqla/models.py]: can we add typing: `def changed_by_name(self) -> str:`

File: superset/connectors/sqla/models.py [hunk 2]
```diff
@@ -440,6 +440,18 @@ def make_sqla_column_compatible(
...
+        if not self.changed_by:
+            return ""
+        return str(self.changed_by)
+
+    @property
+    def changed_by_url(self):
```
Comment (villebro) [inline superset/connectors/sqla/models.py]: here, too: `def changed_by_url(self) -> str:`

**Review by villebro — CHANGES_REQUESTED** [review]
Two type nits

**Review by mistercrunch — APPROVED** [review]

**Comment by villebro** [comment]
When playing around with this, I get the following error with the following filters:
- SQL Lab view (checked)
- Database (empty)

image

**Comment by nytai** [comment]
@villebro Thanks for catching that, I also found that my implementation of the database filter was incorrect. Both issues have been addressed. I've also added an exception which should help catch incorrect filter config moving forward.

File: superset-frontend/src/views/datasetList/DatasetList.tsx [hunk 3]
```diff
@@ -0,0 +1,423 @@
...
+    return Boolean(this.state.permissions.find(p => p === perm));
+  };
+
+  handleDashboardEdit = ({ id }: { id: number }) => {
+    window.location.assign(`/tablemodelview/edit/${id}`);
+  };
+
+  handleDashboardDelete = ({ id, table_name: tableName }: Dataset) =>
+    SupersetClient.delete({
+      endpoint: `/api/v1/dashboard/${id}`,
```
Comment (suddjian) [inline superset-frontend/src/views/datasetList/DatasetList.tsx]: I think you want a different endpoint here

**Review by villebro — APPROVED** [review]
There seems to be some JS lint, otherwise LGTM. This is a *really* nice improvement to the user experience, great work! 👍

**Comment by nytai** [comment]
I just realized this change will remove the create button for datasets(tables). For Charts and Dashboards there was the `NEW` button to fall back on, however that button is missing Tables. This is a blocker, I'll have to figure something out

**Comment by nytai** [comment]
Added a `+` button 

Screen Shot 2020-03-12 at 6 35 10 PM

**Pull request merged**